src/steamship/base/configuration.py

```diff
@@ -45,6 +45,9 @@ class Configuration(CamelModel):
     workspace_handle: str = None
     profile: Optional[str] = None
 
+    # For use in deployed packages and plugins for tracing. Do not set manually
+    request_id: Optional[str] = None
```

Comment:
Not blocking: `Do not set manually` is a surprising thing to read without being as ingrained in the context here yet; it also feels at first blush to me like this is Request-Scoped rather than Client-Scoped.

e.g. in the `RequestIdGenerator` in tests the PluginRequest seems like a natural place this would be, but I'm blanking on a request-scoped place to do this in e.g. a PackageService.

Reply:
Yeah, you're right of course...

The reason I tucked it into the client config object specifically and not the `InvocationContext` was that the client object would need it to send back with every single request to the Engine.  So putting it there allowed me to do it without the package or plugin author ever having to know it existed.

The comment, then, is to say "hey, don't mess with this" since AFAIK there isn't a particularly good Pythonic way of actively preventing someone from changing 'internal' member vars.